thorlcr/master/thgraphmanager.cpp

```diff
@@ -1215,35 +1218,42 @@ void closeThorServerStatus()
  *  0 = unrecognised format, or wuid mismatch
  *  1 = success. new graph/wuid received.
  */
-static int recvNextGraph(unsigned timeoutMs, const char *wuid, StringBuffer &retWuid, StringBuffer &retGraphName)
+static int recvNextGraph(IJobQueue *thorQueue, unsigned timeoutMs, const char *wuid, StringBuffer &retWuid, StringBuffer &retGraphName)
 {
-    PROGLOG("Lingering time left: %.2f", ((float)timeoutMs)/1000);
-    CMessageBuffer msg;
-    if (!queryWorldCommunicator().recv(msg, NULL, MPTAG_THOR, nullptr, timeoutMs))
-        return -1;
     StringBuffer next;
-    msg.read(next);
+    CMessageBuffer msg;
+    if (thorQueue)
+    {
+        Owned<IJobQueueItem> item = thorQueue->dequeue(timeoutMs);
```

Comment:
Question: Is there any way to give the lingering thors a higher priority than waiting eclagents?

Reply:
I know that's a caveat - that it would be better if the lingering Thor instances took Q items, vs an agent (I thought I added a comment somewhere), but I ignored it.
There isn't a way to do that in the IJoqQueue implementation, and didn't think it was worth pursuing.

The net upshot is, that sometimes when the max instances aren't running, there will be times when the agent spins up, up to max when an idle Thor could have taken it instead.

I see this feature mainly being of benefit though to keep the max instances busy, in which case it will not be an issue.
Could open as a separate JIRA if worth looking into further